Archive/Imo/Imo1982Q3.lean

```diff
@@ -0,0 +1,426 @@
+/-
+Copyright (c) 2024 Alex Brodbelt. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex Brodbelt
+-/
+import Mathlib.Algebra.BigOperators.Group.Finset
+import Mathlib.Algebra.Order.Field.Basic
+import Mathlib.Data.NNReal.Basic
+import Mathlib.Algebra.GeomSum
+import Mathlib.Analysis.InnerProductSpace.PiL2
+import Mathlib.Tactic.Linarith
+import Mathlib.Data.Matrix.Basic
+
+open Real BigOperators Finset RealInnerProductSpace Matrix
+
+
+/-!
+# IMO 1982 Q3
+
+Consider infinite sequences $\{x_n \}$ of positive reals such that $x_0 = 0$ and
+$x_0 \geq x_1 \geq x_2 \geq ...$
+
+a) Prove that for every such sequence there is an $n \geq 1$ such that:
+
+$\frac{x_0^2}{x_1} + \ldots + \frac{x_{n-1}^2}{x_n} \geq 3.999$
+
+b) Find such a sequence such that for all n:
+
+$\frac{x_0^2}{x_1} + \ldots + \frac{x_{n-1}^2}{x_n} < 4$
+
+The solution is based on the one at the
+[Art of Problem Solving](https://artofproblemsolving.com/wiki/index.php/1982_IMO_Problems/Problem_3)
+website.
+-/
+
+namespace Imo1982Q3
+
+
+lemma sum_Fin_eq_sum_Ico
+    {x : ℕ → ℝ}
+  :
+  ∀ N,
+  ∑ n : Fin N, x n = ∑ n ∈ Ico 0 N, x n
+  := by
```

Comment:
```suggestion
lemma sum_Fin_eq_sum_Ico {x : ℕ → ℝ} :
  ∀ N, ∑ n : Fin N, x n = ∑ n ∈ Ico 0 N, x n := by
```
Please don't write arguments like this, or at the very least, keep the semicolon in the same line as the last argument previous to it.

Reply:
Ok, I'll keep this in mind. I'll correct the style of other statements as well

Reply:
The latter is more standard throughout Mathlib, but the former should be ok too.

Reply:
Ok, thank you :)